### PR TITLE
Initial build, structure and code for hof-boilerplate

### PR DESCRIPTION
**Pull request by JoeChapman**
A service template for HOF based services. Depends on HOF-bootstrap

Added these assets in this initial commit.
- dotfiles
- assets
- tests dir
- app.js
- meta docs (README..)

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+bootstrap({
+  views: false,
+  fields: false,
+  routes: [{
+    fields: 'any',
+    steps: {
+      '/one': {
+      }
+    }
+  }]
```
Comment (JoeChapman) [inline app.js]: Should probably remove all this config?

**Comment by danielepolencic** [comment]
I'm not sure on what's the policy, but we could use `<svg>`s for icons. Do we still support IE8?

**Comment by danielepolencic** [comment]
A part from the comment above, LGTM 👍

**Comment by joefitter** [comment]
we should remove dotfiles if they are being added postinstall by hof-dotfiles

File: test/common.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+global.chai = require('chai').use(require('sinon-chai'));
+global.should = chai.should();
+global.sinon = require('sinon');
+require('sinomocha')();
+
+process.setMaxListeners(0);
+process.stdout.setMaxListeners(0);
```
Comment (joefitter) [inline test/common.js]: Why is this here? Is it needed?

File: assets/scss/validation.scss [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    text-transform: none;
+    color: #af1324;
+  }
+}
+
+.form-date .form-group.validation-error,
+.panel-indent .form-group.validation-error {
+  border-left: none;
+  padding-left: 0;
+}
```
Comment (joefitter) [inline assets/scss/validation.scss]: Is this stuff HOF specific? Are they not included in a gov stylesheet somewhere?

File: assets/scss/app.scss [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+span.postcode {
+  text-transform: uppercase;
+  font-weight: bold;
+  margin-right: 0.5em;
+}
+
+#address-textarea {
+  height: 6em;
+}
```
Comment (joefitter) [inline assets/scss/app.scss]: How much of this is app-specific? We should look at abstracting anything that is needed for all hof applications

File: assets/scss/alert.scss [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  margin-bottom: .5em;
+  overflow: hidden;
+  vertical-align: middle;
+}
+
+
+.alert-number {
+  @extend .bold-xlarge;
+  float: left;
+  margin-right: .41666667em; /* 20px ÷ 48px font size */
```
Comment (joefitter) [inline assets/scss/alert.scss]: ?

File: assets/countries/index.js [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+  'Western Sahara',
+  'Yemen',
+  'Zambia',
+  'Zimbabwe',
+];
+
+var euCountries = [].concat(countriesDomestic, countriesOtherEU);
+
+module.exports.nonEuCountries = countriesNonEU.sort();
+module.exports.allCountries = [].concat(module.exports.nonEuCountries, euCountries).sort();
```
Comment (joefitter) [inline assets/countries/index.js]: This is for the country typeahead so shouldn't be part of boilerplate. Ideally it should live in its own component

File: assets/js/index.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        };
+
+        var first = compareStartsWithInput(a, b);
+
+        return first === 0 ? compareAlpha(a, b) : first;
+      }
+    }),
+    limit: 100
+  });
+});
```
Comment (joefitter) [inline assets/js/index.js]: See above

File: README.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+Get it `git clone git@github.com:UKHomeOffice/hof-boilerplate.git`
+
+Open `package.json` and edit the `name` and `description` and `repository` fields.
+
+__Hint__: Those fields need to describe elements of the service you are building.
+
+Install everything: `npm install`
+
+Open `app.js` in you editor of choice and start configuring your service.
```
Comment (joefitter) [inline README.md]: remove git history

```bash
rm -rf .git
```

Comment (JoeChapman) [inline README.md]: Is it in fact fine to keep the git history in this case because none of it is specific to an implementation, but only to the tool

File: assets/scss/alert.scss (hunk 5 above)
Comment (JoeChapman) [inline assets/scss/alert.scss]: ?

Comment (JoeChapman) [inline assets/scss/alert.scss]: !

File: test/common.js (hunk 2 above)
Comment (JoeChapman) [inline test/common.js]: I believe it turns off annoying warnings

**Comment by JoeChapman** [comment]
?

File: assets/scss/alert.scss (hunk 5 above)
Comment (JoeChapman) [inline assets/scss/alert.scss]: I actually don't know

**Comment by JoeChapman** [comment]
So, can this be merged?

**Comment by joefitter** [comment]
merging as initial commit, as mentioned any issues can be raised separately
